.github/workflows/milestone.yml

```diff
@@ -0,0 +1,85 @@
+on:
+  pull_request:
+    types:
+      - opened
+      - reopened
+      - closed
+
+permissions:
+  contents: read
+  issues: write
+  pull-requests: write
+
+jobs:
+  set_milestone_label:
+    name: Set milestone label
+    if: ${{ github.event.action != 'closed' && !contains(join(github.event.pull_request.labels.*.name, ','), 'milestone/') }}
+    runs-on: ubuntu-latest
+    steps:
+        - name: Get next minor
+          uses: actions/github-script@5c56fde4671bc2d3592fb0f2c5b5bab9ddae03b1 # 7
+          id: next-milestone
+          with:
+            result-encoding: string
+            script: |
+                const resp = await github.rest.repos.listReleases({
+                    ...context.repo,
+                });
+                const releases = resp.data;
+                const latestVersion = releases.filter(x => x.tag_name.startsWith("v6.7"))[0].tag_name;
+                const splittedVersion = latestVersion.split('.');
+                const major = parseInt(splittedVersion[1]);
+                const minor = parseInt(splittedVersion[2]);
+                const patch = parseInt(splittedVersion[3]);
+
+                const versionRegex = new RegExp(/^([0-9]+).([0-9]+).([0-9]+).([0-9]+|x)$/, "mg");
+
+                // Increase minor if target is trunk or is an issue
+                if (context.payload.pull_request.base.ref == "trunk") {
+                  return `6.${major}.${minor+1}.0`
+                // Increase patch if target is a version branch
+                } else if (versionRegex.test(context.payload.pull_request.base.ref)) {
+                  return `6.${major}.${minor}.${patch+1}`
+                }
+
+                // Skip if the PR doesn't target trunk or an version branch
+                core.warning("This PR doesn't target 'trunk' or a version branch");
+                return "skip";
```

Comment:
[This action](https://github.com/shopware/github-actions/blob/main/versions/action.yml), I believe, already handles parts of this step. It might make sense to reference it and optionally implement the additional logic for upcoming versions there